modules/download_ec2_userdata/main.py

```diff
@@ -1,119 +1,119 @@
-                region,
-                base64.b64decode(user_data['Value'])
-            )
-            print('  {}@{} user data found'.format(instance_id, region))
-
-            with open('sessions/{}/downloads/user_data.txt'.format(session.name), 'a+') as data_file:
-                data_file.write(formatted_user_data)
-            summary_data['userdata_downloads'] += 1
-
-        else:
-            print('  {}@{} user data not found'.format(instance_id, region))
-
-    print('\n{} completed.\n'.format(module_info['name']))
-    return summary_data
-
-
-def summary(data, pacu_main):
-    out = ''
-    if 'dl_path'in data:
-        out += '  Data downloaded to: \n    {}\n'.format(data['dl_path'])
-    out += '  {} instance(s) user data downloaded'.format(data['userdata_downloads'])
-    return out
+#!/usr/bin/env python3
+import argparse
+import base64
+import os
+
+from botocore.exceptions import ClientError
+
+
+module_info = {
+    # Name of the module (should be the same as the filename)
+    'name': 'download_ec2_userdata',
+
+    # Name and any other notes about the author
+    'author': 'Spencer Gietzen of Rhino Security Labs',
+
+    # Category of the module. Make sure the name matches an existing category.
+    'category': 'recon_enum_with_keys',
+
+    # One liner description of the module functionality. This shows up when a user searches for modules.
+    'one_liner': 'Downloads user data from EC2 instances.',
+
+    # Description about what the module does and how it works
+    'description': 'This module will take a list of EC2 instance IDs and request then download the User Data associated with each instance. All of the data will be saved to ./sessions/[session_name]/downloads/user_data.txt.',
+
+    # A list of AWS services that the module utilizes during its execution
+    'services': ['EC2'],
+
+    # For prerequisite modules, try and see if any existing modules return the data that is required for your module before writing that code yourself, that way, session data can stay separated and modular.
+    'prerequisite_modules': ['enum_ec2'],
+
+    # Module arguments to autocomplete when the user hits tab
+    'arguments_to_autocomplete': ['--instance-ids'],
+}
+
+parser = argparse.ArgumentParser(add_help=False, description=module_info['description'])
+
+parser.add_argument('--instance-ids', required=False, default=None, help='One or more (comma separated) EC2 instance IDs with their regions in the format instance_id@region. Defaults to all EC2 instances.')
+
+
+def main(args, pacu_main):
+    session = pacu_main.get_active_session()
+
+    ###### Don't modify these. They can be removed if you are not using the function.
+    args = parser.parse_args(args)
+    print = pacu_main.print
+    fetch_data = pacu_main.fetch_data
+    ######
+
+    instances = []
+    summary_data = {'userdata_downloads': 0}
+    # Check permissions before doing anything
+    try:
+        client = pacu_main.get_boto3_client('ec2', 'us-east-1')

```

Comment:
This should get changed to be a region from get_regions() incase us-east-1 isn't in the session regions